irohad/maintenance/metrics.cpp

```diff
@@ -4,93 +4,139 @@
+                                .Register(*registry_);
+  auto&block_height = block_height_gauge.Add({});//{{"value", "some"}});
+  block_height.Set(storage_->getBlockQuery()->getTopBlockHeight());
 
-      if (random_value & 1) tcp_rx_counter.Increment();
-      if (random_value & 2) tcp_tx_counter.Increment();
-      if (random_value & 4) udp_rx_counter.Increment();
-      if (random_value & 8) udp_tx_counter.Increment();
+  auto&peers_number_gauge = BuildGauge()
+      .Name("peers_number")
+      .Help("Total number peers to send transactions and request proposals")
+      //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&number_of_peers = peers_number_gauge.Add({});//{{"valueP", "any"}});
 
-      const std::array<std::string, 4> methods = {"GET", "PUT", "POST", "HEAD"};
-      auto method = methods.at(random_value % methods.size());
-      // dynamically calling Family<T>.Add() works but is slow and should be avoided
-      http_requests_counter.Add({{"method", method}}).Increment();
-    }
-  }).detach();
+  auto&domains_number_gauge = BuildGauge()
+      .Name("number_of_domains")
+      .Help("Total number of domains in WSV")
+      //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&domains_number = domains_number_gauge.Add({});
 
-  return {registry};
+  auto&total_number_of_transactions_gauge = BuildGauge()
+      .Name("number_of_signatures_in_last_block")
+      .Help("Number of signatures in last block")
+          //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&total_number_of_transactions = total_number_of_transactions_gauge.Add({});
+  
+  auto&number_of_signatures_in_last_block_gauge = BuildGauge()
+      .Name("number_of_signatures_in_last_block")
+      .Help("Number of signatures in last block")
+          //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&number_of_signatures_in_last_block = number_of_signatures_in_last_block_gauge.Add({});
+  
+  block_subscriber_ = std::make_shared<BlockSubscriber>(
+      getSubscription()->getEngine<EventTypes,BlockPtr>());
+  block_subscriber_->setCallback(
+        [&] //Values are stored in registry_
+        (auto, auto&receiver, auto const event, BlockPtr pblock){
+          // block_height is captured by reference because it is stored inside registry_, which is shared_ptr
+          assert(pblock);
+          block_height.Set(pblock->height());
+          ///---
+          int domains_diff = 0, peers_diff=0;
+          unsigned signatures_num = 0;
+          using namespace shared_model::interface;
+          for(Transaction const& trx : pblock->transactions()){
+            for(Command const& cmd : trx.commands()){
+              domains_diff += command_is<CreateDomain>(cmd);
+              //do it later: domains_diff -= boost::get<const&RemoveDomain>(&cmd.get()) != nullptr;
+              peers_diff += command_is<AddPeer>(cmd);
+              peers_diff -= command_is<RemovePeer>(cmd);
```

Comment:
I understand what is happening here, but it looks so strange:) The first think was to check types overflows while calculating. It seems better to do smth like:
```c++
domains_diff += cmd.is<CreateDomain>() ? 1 : 0;

or even:

{...}.match(
[&domains_diff](CreateDomain const &) {
   ++domains_diff;
}, [&peers_diff](AddPeer const &) {
   ++peers_diff;
}, [&peers_diff](RemovePeer const &) {
   --peers_diff;
});

Reply:
1st variant I accept. 2nd one has toooo much letters

Reply:
And really bool for math operations converted to integer implicitly.